### PR TITLE
Mastery Spec added to Legend Engine

### PR DESCRIPTION
**Pull request by christian-gallagher**
Add Mastery specification to Legend Engine including the PURE, protocol and grammar modules. All tests pass and it has been tested in conjunction with the Studio changes in MR - Master Data Management Specification studio #1404.

### CODE REVIEW

**Comment by christian-gallagher** [comment]
First commit was without an email client, I have rebased to correct this, now just pending approval.

File: legend-engine-server/pom.xml [hunk 1]
```diff
@@ -162,6 +162,16 @@
             <artifactId>legend-engine-xt-persistence-cloud-grammar</artifactId>
             <scope>runtime</scope>
         </dependency>
+        <dependency>
```
Comment (epsstan) [inline legend-engine-server/pom.xml:165]: I don't see any new code being added to the server module. Is this dependency required ?

File: legend-engine-xt-mastery-pure/src/test/java/org/finos/legend/pure/code/core/mastery/Test_Pure_Mastery.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+//  limitations under the License.
+
+package org.finos.legend.pure.code.core.mastery;
+
+import junit.framework.TestSuite;
+import org.finos.legend.pure.code.core.compiled.test.PureTestBuilderHelper;
+import org.finos.legend.pure.m3.execution.test.TestCollection;
+import org.finos.legend.pure.runtime.java.compiled.execution.CompiledExecutionSupport;
+
+public class Test_Pure_Mastery
```
Comment (epsstan) [inline legend-engine-xt-mastery-pure/src/test/java/org/finos/legend/pure/code/core/mastery/Test_Pure_Mastery.java:22]: The tests have been commented out. Is this deliberate ?

File: legend-engine-xt-mastery-grammar/src/test/java/org/finos/legend/engine/language/pure/dsl/mastery/grammar/test/TestMasteryGrammarParser.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    @Override
+    public Vocabulary getParserGrammarVocabulary()
+    {
+        return MasteryParserGrammar.VOCABULARY;
+    }
+
+    @Override
+    public String getParserGrammarIdentifierInclusionTestCode(List<String> keywords)
+    {
+        System.out.println(TestMasteryCompilationFromGrammar.COMPLETE_CORRECT_MASTERY_MODEL);
```
Comment (epsstan) [inline legend-engine-xt-mastery-grammar/src/test/java/org/finos/legend/engine/language/pure/dsl/mastery/grammar/test/TestMasteryGrammarParser.java:35]: Nitpick. Remove sysout

**Comment by epsstan** [comment]
+ 1 for comprehensive doc tags!

Apart from the addition of a new Maven dependency to the server module, this looks like mostly boilerplate code to introduce a new DSL.  Has the DSL by itself been reviewed by @pierredebelen /others ?

File: legend-engine-server/pom.xml (hunk 1 above)
Comment (christian-gallagher) [reply legend-engine-server/pom.xml:165]: It is not, I have removed it.

File: legend-engine-xt-mastery-pure/src/test/java/org/finos/legend/pure/code/core/mastery/Test_Pure_Mastery.java (hunk 2 above)
Comment (christian-gallagher) [reply legend-engine-xt-mastery-pure/src/test/java/org/finos/legend/pure/code/core/mastery/Test_Pure_Mastery.java:22]: It is intentional, I clarified with a comment.

File: legend-engine-xt-mastery-grammar/src/test/java/org/finos/legend/engine/language/pure/dsl/mastery/grammar/test/TestMasteryGrammarParser.java (hunk 3 above)
Comment (christian-gallagher) [reply legend-engine-xt-mastery-grammar/src/test/java/org/finos/legend/engine/language/pure/dsl/mastery/grammar/test/TestMasteryGrammarParser.java]: Done - missed it!

**Comment by christian-gallagher** [comment]
> * 1 for comprehensive doc tags!
> 
> Apart from the addition of a new Maven dependency to the server module, this looks like mostly boilerplate code to introduce a new DSL. Has the DSL by itself been reviewed by @pierredebelen /others ?

Yes - this has been reviewed in detail with Dave Lee and Pierre agreed to approve the addition on Friday 5th Aug.

**Pull request merged**